libraries/helpers.rb

```diff
@@ -43,6 +43,13 @@ def local_package_resource
     def package_resource(ingredient_action)
       presource = new_resource.package_source.nil? ? :package : local_package_resource
 
+      declare_resource :ruby_block, 'stop chef run' do
+        action :nothing
+        block do
+          fail 'Chef version has changed during the run. Stopping the current Chef run. Please run chef again.'
```

Comment:
It may be worth writing a bit more verbose messaging here (a la the comment below?), at `Chef::Log.fatal` level, and using `Chef::Application.fatal!`.